controllers/csiaddonsnode_controller.go

```diff
@@ -58,5 +121,50 @@ func (r *CSIAddonsNodeReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 func (r *CSIAddonsNodeReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&csiaddonsv1alpha1.CSIAddonsNode{}).
+		WithEventFilter(predicate.GenerationChangedPredicate{}).
 		Complete(r)
 }
+
+// addFinalizer adds finalizer to csiAddonsNode if it is not present.
+func (r *CSIAddonsNodeReconciler) addFinalizer(
```

Comment:
With this, each controller will need its own `addFinalizer` function?

- If yes, we can leave it as it is and define separate functions in each controller.

- But if we want a generic one, better to keep in a new `finalizers.go`

Thoughts? @Rakshith-R @nixpanic @Madhu-1